File: app/src/main/java/com/vocaby/app/database/entity/Word.java
package com.vocaby.app.database.entity;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "dictionary_word")
public class Word {
    @PrimaryKey(autoGenerate = true)
    private int id;

    @NonNull
    private String word;
    private String pronunciation;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getWord() {
        return word;
    }

    public void setWord(String word) {
        this.word = word;
    }

    public void setPronunciation(String pronunciation) {
        this.pronunciation = pronunciation;
    }

    public String getPronunciation() {
        return pronunciation;
    }
}

File: app/src/main/java/com/vocaby/application/feature_dictionary/data/local/entity/Word.kt
package com.vocaby.application.feature_dictionary.data.local.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import java.util.*

@Entity(tableName = "dictionary_word", indices = [Index(value=["word", "id"])])
data class Word constructor(
    var word: String = "",
    var pronunciation:String? = null,
    @ColumnInfo(name = "last_updated")
    var lastUpdated: Date = Date(),
    @PrimaryKey(autoGenerate = true) var id: Int = 0
)